cmake/Dependencies.cmake

```diff
@@ -1156,6 +1156,18 @@ if(APPLE)
   target_link_options(pybind::pybind11 INTERFACE -undefined dynamic_lookup)
 endif()
 
+# ---[ OpenTelemetry API headers
+if(NOT OpenTelemetryApi_FOUND)
+  message(STATUS "Using third_party/opentelemetry-cpp.")
+  set(OpenTelemetryApi_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR}/../third_party/opentelemetry-cpp/api/include)
+  install(DIRECTORY ${OpenTelemetryApi_INCLUDE_DIRS}
+          DESTINATION ${CMAKE_INSTALL_PREFIX}
+          FILES_MATCHING PATTERN "*.h")
```

Comment:
If possible, I'd like to avoid leaking open telemetry headers into our public headers. If they aren't leaked into public headers, then it is not necessary to install the opentelemetry headers with the rest of our C++ headers. This makes it easier to interoperate with other C++ codebases which may want to use their own version of opentelemetry.

Reply:
this shouldn't block landing the PR

Reply:
Yes, it's possible not to do the install step. Thanks for the suggestion. I removed this and things are still fine.